subprojects/composite-builds/src/integTest/groovy/org/gradle/integtests/composite/AbstractCompositeBuildIntegrationTest.groovy

```diff
@@ -132,6 +134,69 @@ abstract class AbstractCompositeBuildIntegrationTest extends AbstractIntegration
         assert operations.roots().size() == 1
     }
 
+    class TimePoint implements Comparable<TimePoint> {
+        private final boolean end
+        private final long time
+        private final BuildOperationRecord operation
+
+        TimePoint(BuildOperationRecord operation, long time) {
+            this(operation, time, false)
+        }
+
+        TimePoint(BuildOperationRecord operation, long time, boolean end) {
+            this.operation = operation
+            this.time = time
+            this.end = end
+        }
+
+        @Override
+        int compareTo(TimePoint o) {
+            if (o.time > time) return -1
+            else if (o.time < time) return 1
+            else {
+                if (end) {
+                    return -1
+                } else {
+                    return 0
+                }
+            }
+        }
+
+        @Override
+        String toString() {
+            if (end) time + "E"
+            else time + "S"
+        }
+    }
+
+    void assertConcurrentOperationsDoNotExceed(Class<BuildOperationType> type, int maximumConcurrentOperations, boolean concurrencyExpected=false) {
```

Comment:
Oooh, this is shiny...Might be a cool thing to add to `BuildOperationsFixture` so it can be used anywhere.